# 연습문제1-1
사용자로부터 정수를 입력받아  
해당 정수의 크기만큼 '안녕'을 출력하세요.


In [2]:
a = int(input("정수를 입력해주세요:"))

# 코드 작성
for i in range(a):
    print("안녕")


정수를 입력해주세요:5
안녕
안녕
안녕
안녕
안녕


# 연습문제1-2
사용자로부터 정수를 입력받아  
사용자로부터 정수를 입력받아, 입력된 정수 만큼 별 찍기


In [3]:
a = int(input("정수를 입력해주세요:"))


for i in range(1, a+1):
    print('*' * (i))

# # 또는 다음도 가능
# for i in range(a):
#     print('*'*(i+1))

# # for문 두개로는
# for i in range(1, a+1):
#     star = "*"
#     for x in range(1, i):
#         star+="*"
#     print(star)

정수를 입력해주세요:5
*
**
***
****
*****


# 연습문제1-3
사용자로부터 정수를 입력받아  
사용자로부터 정수를 입력받아, 입력된 정수 만큼 별 찍기 **(역순)**


In [5]:
a = int(input("정수를 입력해주세요:"))

for i in range(a, 0, -1):
    print('*' * (i))

정수를 입력해주세요:5
*****
****
***
**
*


# 연습문제2
------

# 연습문제2-1
(1)  x = [3, 6, 9, 20, -7, 5] 의 값의 모든 요소에 10을 곱한뒤 출력하세요  
**심화 : 출력과 리스트 x 의 값에도 모두 10을 곱해주세요**


In [26]:
x = [3, 6, 9, 20, -7, 5]

# 코드 작성
for item in x:
    print(item * 10)
    
    
####### 심화 #######
# for idx, item in enumerate(x):
#     x[idx] *= 10
#     print(x[idx])

# print('-'*10) # 구분선
# print(x)


30
60
90
200
-70
50
30
60
90
200
-70
50
----------
[30, 60, 90, 200, -70, 50]


# 연습문제2-2
y = {“math”: 70, “science”: 80, “english”: 20} 의 값의 모든 요소에 10을 더한뒤 출력하세요  
**심화 : 출력과 딕셔너리 y 의 값에도 모두 10을 더해주세요**


In [25]:
y = {'math': 70, 'science': 80, 'english': 20}

# 코드 작성
for subject in y:
    print(y[subject]+10)


####### 심화 #######
for subject in y:
    y[subject] +=10
    print(y[subject])

# print('-'*10) # 구분선
# print(y)

80
90
30
80
90
30
----------
{'math': 80, 'science': 90, 'english': 30}


# 연습문제2-3
숫자를 입력받고 입력받은 정수의 구구단을 출력하세요


In [7]:
a = int(input("정수를 입력하세요: "))
# 코드 작성

for x in range(1, 10):
    print("{0} * {1} = {2}".format(a, x, a*x))
    print(a, '*', x, '=', a*x)
    print(f'{a} * {x} = {a*x}')
    # or
    # print("%s * %s = %s" % (a,x, a*x))
    

정수를 입력하세요: 5
5 * 1 = 5
5 * 1 = 5
5 * 1 = 5
5 * 2 = 10
5 * 2 = 10
5 * 2 = 10
5 * 3 = 15
5 * 3 = 15
5 * 3 = 15
5 * 4 = 20
5 * 4 = 20
5 * 4 = 20
5 * 5 = 25
5 * 5 = 25
5 * 5 = 25
5 * 6 = 30
5 * 6 = 30
5 * 6 = 30
5 * 7 = 35
5 * 7 = 35
5 * 7 = 35
5 * 8 = 40
5 * 8 = 40
5 * 8 = 40
5 * 9 = 45
5 * 9 = 45
5 * 9 = 45


# 연습문제3-1
word = [“school”, “game”, “piano”, “science”, “hotel”, “mountain”]   
중 글자수가 6글자 이상인 문자를 모아 새로운 리스트를 생성하세요

In [28]:
word = ['school', 'game', 'piano', 'science', 'hotel', 'mountain']

# 코드 작성
new_word = [] # or list()

for x in word:
    if len(x)>=6:
        new_word.append(x)

print(new_word)



['school', 'science', 'mountain']


In [11]:
word = ['school', 'game', 'piano', 'science', 'hotel', 'mountain']

# 코드 작성
new_word = [] # or list()

for x in word:
    if len(x)>=6:
        new_word += [x]
        # new_word.extend([x])

print(new_word)



['school', 'science', 'mountain']


In [9]:
word = ['school', 'game', 'piano', 'science', 'hotel', 'mountain']

for x in word:
    print(x)
    
    if len(x)<6:
        word.remove(x)

school
game
science
hotel


In [8]:
word = ['school', 'game', 'piano', 'science', 'hotel', 'mountain']
new_word = word.copy()

for x in word:
    if len(x)<6:
        new_word.remove(x)
        
new_word

['school', 'science', 'mountain']

# 연습문제3-2
구구단을 1단부터 9단까지 출력하세요

In [12]:
# 코드 작성

for i in range(2, 10):
    for j in range(1, 10):
        print("{0} * {1} = {2}".format(i, j, i*j))
        # or
        # print("%s * %s = %s" % (i,j, i*j))
    print("-"* 10, "%s단 끝" % i)
    

2 * 1 = 2
2 * 2 = 4
2 * 3 = 6
2 * 4 = 8
2 * 5 = 10
2 * 6 = 12
2 * 7 = 14
2 * 8 = 16
2 * 9 = 18
---------- 2단 끝
3 * 1 = 3
3 * 2 = 6
3 * 3 = 9
3 * 4 = 12
3 * 5 = 15
3 * 6 = 18
3 * 7 = 21
3 * 8 = 24
3 * 9 = 27
---------- 3단 끝
4 * 1 = 4
4 * 2 = 8
4 * 3 = 12
4 * 4 = 16
4 * 5 = 20
4 * 6 = 24
4 * 7 = 28
4 * 8 = 32
4 * 9 = 36
---------- 4단 끝
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
---------- 5단 끝
6 * 1 = 6
6 * 2 = 12
6 * 3 = 18
6 * 4 = 24
6 * 5 = 30
6 * 6 = 36
6 * 7 = 42
6 * 8 = 48
6 * 9 = 54
---------- 6단 끝
7 * 1 = 7
7 * 2 = 14
7 * 3 = 21
7 * 4 = 28
7 * 5 = 35
7 * 6 = 42
7 * 7 = 49
7 * 8 = 56
7 * 9 = 63
---------- 7단 끝
8 * 1 = 8
8 * 2 = 16
8 * 3 = 24
8 * 4 = 32
8 * 5 = 40
8 * 6 = 48
8 * 7 = 56
8 * 8 = 64
8 * 9 = 72
---------- 8단 끝
9 * 1 = 9
9 * 2 = 18
9 * 3 = 27
9 * 4 = 36
9 * 5 = 45
9 * 6 = 54
9 * 7 = 63
9 * 8 = 72
9 * 9 = 81
---------- 9단 끝


# 연습문제4
**1~100** 까지 숫자중  
3과 5의 **공배수**일경우 “3과 5의 공배수”  
그 외 **3의배수**일경우 “3의배수”  
그 외 **5의배수**일경우 “5의배수”  
모두 해당되지 않을경우 그냥숫자를 출력하세요  

**심화 : 1~입력한숫자까지의 숫자중**


In [16]:
# for a in range(1, 101):
#     if a%3==0 and a%5==0: # 먼저 나와야 함.
#         print("3과 5의 공배수")
#     elif a%3==0:
#         print("3의 배수")
#     elif a%5==0:
#         print("5의 배수")
#     else:
#         print(a)
        
##### 심화 #####    
num = int(input("정수 입력하세요. "))
for a in range(1, num+1):
    if a%3==0 and a%5==0: # 먼저 나와야 함.
        print("3과 5의 공배수")
    elif a%3==0:
        print("3의 배수")
    elif a%5==0:
        print("5의 배수")
    else:
        print("그냥 숫자")

정수 입력하세요. 1000
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 숫자
3의 배수
5의 배수
그냥 숫자
3의 배수
그냥 숫자
그냥 숫자
3과 5의 공배수
그냥 숫자
그냥 숫자
3의 배수
그냥 숫자
5의 배수
3의 배수
그냥 숫자
그냥 

-----
# while 문

# 연습문제 5

사용자로부터 숫자를 계속 입력받다가  
s or S 를 입력하면 합계출력


In [18]:
# 코드 작성
number = 0
total = 0

while str(number).lower() != "s":
    total += int(number)
    number = input("값을 입력해주세요")

print("합계는 ? ", total)

값을 입력해주세요10
값을 입력해주세요1
값을 입력해주세요2
값을 입력해주세요3
값을 입력해주세요s
합계는 ?  16


# 연습문제 6
가위바위보 게임

In [6]:
# import random
# user = 0
# total = 0
# win = 0
# word = ["가위", "바위", "보"]

# user = int(input("가위(1), 바위(2), 보(3) 을 입력해주세요! : "))
# computer = random.randint(1, 3)

# while user in [1, 2, 3]:
#     print("유저 : %s, 컴퓨터 : %s" % (word[user - 1], word[computer - 1]) )

#     if(user==1 and computer==3):
#         win += 1
#     elif(user==2 and computer==1):
#         win += 1
#     elif(user==3 and computer==2):
#         win += 1
        
#     total +=1
    
#     print('-'*10)
#     user = int(input("가위(1), 바위(2), 보(3) 을 입력해주세요! : "))
#     computer = random.randint(1, 3)

# print('='*10)
# print("게임종료 (전체:{} ,승리:{})".format(total, win))    



###### 심화 ######
import random

user = 0
total = 0
win = 0
draw = 0

word = ["가위", "바위", "보"]

# 1-

user = int(input("가위(1), 바위(2), 보(3) 을 입력해주세요! : "))
computer = random.randint(1, 3)

while user in [1, 2, 3]: 
    print("유저 : %s, 컴퓨터 : %s" % (word[user - 1], word[computer - 1]) )

    # 승리 체크
    if(user==1 and computer==3): # user:가위 computer: 보
        win += 1
    elif(user==2 and computer==1):
        win += 1
    elif(user==3 and computer==2):
        win += 1        
    elif (user==computer): # 비긴숫자
        draw += 1 
        
    total += 1
    
    print('-'*10)
    user = int(input("가위(1), 바위(2), 보(3) 을 입력해주세요! : "))
    computer = random.randint(1, 3)

print('='*10)
print("게임종료 (전체:{} ,승리:{}, 비김:{})".format(total, win, draw))    

가위(1), 바위(2), 보(3) 을 입력해주세요! : 3
유저 : 보, 컴퓨터 : 보
----------
가위(1), 바위(2), 보(3) 을 입력해주세요! : 3
유저 : 보, 컴퓨터 : 가위
----------
가위(1), 바위(2), 보(3) 을 입력해주세요! : 5
게임종료 (전체:2 ,승리:0, 비김:1)


# 연습문제 7
randint함수를 사용하여 정수(1~1000)로 이루어진 길이가 100개인 리스트 a를 만들고   
가장 큰 수를 구하는 코드를 작성하시오.  

예) a=[32, 45, 2, 5, 76] 일때, 출력결과는 76이 나와야 함.  





In [12]:
import random
a = []
# 코드 작성

for i in range(100):
    a.append(random.randint(1,1000))

print("list:", a)

# [1, 5, 8, 3, 2]

# max_a = a[0]
# for i in a:
#     if max_a<i:
#         max_a = i

# print("가장 큰 수:", max_a)


# ## 또는
# max_a = max(a)
# print("가장 큰 수:", max_a)

# ## 또는
# max_a = a.sort()
a.sort(reverse=True) # 정렬후에
max_a = a[0]
print("정렬후 list: ", a)
print("가장 큰 수:", max_a)



list: [961, 212, 921, 427, 432, 648, 413, 72, 423, 908, 727, 950, 863, 565, 233, 963, 644, 498, 859, 794, 387, 776, 14, 84, 21, 153, 956, 917, 782, 891, 675, 125, 156, 959, 981, 620, 973, 979, 769, 632, 353, 560, 808, 542, 278, 65, 21, 232, 645, 810, 355, 348, 651, 551, 414, 978, 235, 378, 436, 189, 637, 830, 951, 546, 956, 272, 743, 122, 900, 737, 270, 86, 12, 878, 358, 770, 876, 80, 503, 547, 193, 850, 522, 886, 39, 431, 504, 854, 176, 290, 301, 574, 763, 138, 406, 190, 794, 844, 177, 740]
정렬후 list:  [981, 979, 978, 973, 963, 961, 959, 956, 956, 951, 950, 921, 917, 908, 900, 891, 886, 878, 876, 863, 859, 854, 850, 844, 830, 810, 808, 794, 794, 782, 776, 770, 769, 763, 743, 740, 737, 727, 675, 651, 648, 645, 644, 637, 632, 620, 574, 565, 560, 551, 547, 546, 542, 522, 504, 503, 498, 436, 432, 431, 427, 423, 414, 413, 406, 387, 378, 358, 355, 353, 348, 301, 290, 278, 272, 270, 235, 233, 232, 212, 193, 190, 189, 177, 176, 156, 153, 138, 125, 122, 86, 84, 80, 72, 65, 39, 21, 21, 14, 12]
가

In [7]:
import random

a = []

for x in range(100):
    a.append(random.randint(1,1000))
    
print(a)

[1, 743, 348, 509, 576, 979, 199, 993, 724, 663, 628, 519, 23, 196, 634, 133, 545, 261, 660, 912, 290, 184, 421, 530, 44, 330, 394, 219, 605, 827, 187, 239, 219, 405, 207, 211, 232, 733, 321, 999, 312, 114, 899, 594, 79, 487, 469, 347, 556, 518, 703, 415, 949, 160, 515, 510, 527, 264, 222, 708, 106, 600, 246, 188, 45, 922, 738, 400, 753, 277, 559, 88, 281, 530, 662, 869, 445, 437, 830, 447, 860, 298, 479, 469, 197, 122, 738, 266, 172, 71, 198, 572, 653, 370, 414, 157, 65, 854, 784, 582]


# 연습문제 8
randint함수를 사용하여 정수(1~1000)로 이루어진 길이가 100개인 리스트 a를 만들고 
리스트의 모든 요소의 합계를 구하는 코드를 작성하시오.

예) a=[32, 45, 2, 5, 76] 일때, 출력결과는 160이 나와야 함.







In [13]:
import random
a = []
# 코드 작성

for i in range(100):
    a.append(random.randint(1,1000))

print("list:", a)

sum_a = 0
for i in a:
    sum_a += i

print("합계:", sum_a)

## 또는
# print("합계:", sum(a))
sum(a)

list: [436, 222, 795, 159, 308, 985, 900, 659, 406, 489, 869, 815, 428, 289, 436, 836, 328, 965, 465, 417, 30, 598, 800, 413, 850, 151, 266, 218, 649, 817, 556, 943, 941, 911, 369, 73, 468, 787, 750, 118, 899, 900, 672, 378, 898, 738, 851, 5, 944, 855, 29, 585, 152, 193, 559, 471, 458, 21, 527, 83, 122, 704, 477, 66, 125, 646, 302, 446, 786, 791, 930, 355, 503, 558, 320, 966, 662, 458, 978, 944, 25, 675, 238, 489, 701, 671, 364, 516, 381, 742, 652, 13, 86, 800, 694, 703, 470, 621, 510, 254]
합계: 52877


52877

# 연습문제 9-1
word = [“school”, “game”, “piano”, “science”, “hotel”, “mountain”] 중   
글자수가 6글자 이상인 문자를 모아 새로운 리스트를 생성하세요  
**(리스트 컴프리핸션을 사용해주세요)**


In [14]:
word = ['school', 'game', 'piano', 'science', 'hotel', 'mountain']

# 코드 작성
new_word = [w for w in word if len(w) >=6]
print(new_word)

['school', 'science', 'mountain']


# 연습문제 9-2
word = [“school”, “game”, “piano”, “science”, “hotel”, “mountian”] 리스트의  
글자수가 들어가있는 새로운 리스트를 생성하세요  
**(리스트 컴프리핸션을 사용해주세요)**



In [69]:
word = ['school', 'game', 'piano', 'science', 'hotel', 'mountain']

# 코드 작성
new_word = [len(w) for w in word]
print(new_word)

[6, 4, 5, 7, 5, 8]


# 연습문제 9 

In [15]:
a = [[10, 20], [30, 40], [50, 60]]
b = [[2, 3], [4, 5], [6, 7]]

# [[20, 60], [120, 200], [300, 420]]


In [23]:
len(a)

3

In [24]:
len(b)

3

In [ ]:
append( ), extend( )

In [48]:
a = [1, 2, 3, 4]


In [49]:
a.append([5, 3])
# a.append({'key': 'value'})
# a

[1, 2, 3, 4, [5, 6], [4, 2]]

In [50]:
a = [[10, 20], [30, 40], [50, 60]]
b = [[2, 3], [4, 5], [6, 7]]

# [[20, 60], [120, 200], [300, 420]]

In [ ]:
c = []
# for
# 1. for문을 통해서 큰 리스트 인덱스를 가져온다.
for i in range(len(a)):
    # 2. 각 인덱스를 통해서 a리스트와 b리스트의 
    # 작은 리스트의 값을 가져온다.
#     small_a = a[i]
#     s_a_0 = a[i][0]
#     s_a_1 = a[i][1]
# #     print(small_a)
# #     print(s_a_0)
# #     print(s_a_1)
#     small_b = b[i]
#     s_b_0 = b[i][0]
#     s_b_1 = b[i][1]
    
    c.append([a[i][0]*b[i][0], a[i][1] * b[i][1]])
#     print(small_b)
#     print(s_b_0)
#     print(s_b_1)
#     print("-"*10)
    # 3. a리스트와 b리스트의 값을 곱하여 
    # [a[i] * b[i], a[i+1] *b[i+1]] 추가하겠다.
    
print(c)
    
    
    

# [[20, 60], [120, 200], [300, 420]]

In [60]:
a = [[10, 20, 30, 50], [30, 40, 30], [50, 60]]
b = [[2, 3, 30, 50], [4, 5, 30], [6, 7]]

# c = [[20, 60, 900]. [120, 200, 900], [300, 420]]

In [62]:
c = []
# for
# 1. for문을 통해서 큰 리스트 인덱스를 가져온다.
d = []
for i in range(len(a)):
    #  a[0] = [10, 20, 30, 50]
    #  b[0] = [2, 3, 30, 50]
    
    #  a[1] = [30, 40, 30]
    #  b[1] = [4, 5, 30]
    d = []
    # 2. 작은(두번째) 리스트의 인덱스를 for문은로 가져온다.

    for j in range(len(a[i])): # [10, 20, 30]
        # [30, 40, 30]
        d.append(a[i][j] * b[i][j])
        

    c.append(d)
    # c[0] = [20, 60, 900, 2500]
print(c)

# print(c)
    
#     c.append([a[i][0]*b[i][0], a[i][1] * b[i][1]])

#     print(small_b)
#     print(s_b_0)
#     print(s_b_1)
#     print("-"*10)
    # 3. a리스트와 b리스트의 값을 곱하여 
    # [a[i] * b[i], a[i+1] *b[i+1]] 추가하겠다.
    
# print(c)
    
    
    
# c = [[20, 60, 900]. [120, 200, 900], [300, 420]]

[[20, 60, 900, 2500], [120, 200, 900], [300, 420]]


In [64]:
a = [[10, 20, 30, 50], [30, 40, 30], [50, 60]]
b = [[2, 3, 30, 50], [4, 5, 30], [6, 7]]

# c = [[20, 60, 900]. [120, 200, 900], [300, 420]]

In [67]:
c = [a[i][j]*b[i][j] for i in range(len(a)) for j in range(len(a[i]))]
c

[[20, 120, 300], [60, 200, 420]]

In [68]:
c = [ [ a[i][j]*b[i][j] for j in range(len(a[i])) ] for i in range(len(a)) ]
c

[[20, 60, 900, 2500], [120, 200, 900], [300, 420]]

In [63]:


for i in range(len(a)):
    #  a[0] = [10, 20, 30, 50]
    #  b[0] = [2, 3, 30, 50]
    
    #  a[1] = [30, 40, 30]
    #  b[1] = [4, 5, 30]
    d = []
    # 2. 작은(두번째) 리스트의 인덱스를 for문은로 가져온다.

    for j in range(len(a[i])): # [10, 20, 30]
        # [30, 40, 30]
        d.append(a[i][j] * b[i][j])
        

    c.append(d)

In [ ]:
# for

c = []
for i in range(len(a)): # 0, 1, 2
    

In [20]:
# zip

In [19]:
# for 

In [18]:
for i in a:
    print(i)

[10, 20]
[30, 40]
[50, 60]


# 연습문제 11
(1) while 문과 break 문을 사용하여 사용자가 입력한 숫자만큼 반복하여 ‘hi’ 출력하기




In [70]:
num = int(input("얼마나 인사를 할까요? "))
count = 0

while True:
    if num == count:
        break
    
    print("hi")
    
    count += 1

얼마나 인사를 할까요? 5
hi
hi
hi
hi
hi


(2) for 문과 continue 문을 사용하여 사용자가 입력한 숫자까지의 짝수를 출력하기

In [71]:
num = int(input("어디까지 짝수를 출력할까요?"))
for i in range(num+1):
    if i%2==1:
        continue
    print(i)

어디까지 짝수를 출력할까요?100
0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100
